index.bs

```diff
@@ -8972,6 +9018,373 @@ worker=] algorithm:
+
+    1. Return false.
+
+1. Return true.
+
+</div>
+
+<div algorithm>
+To <dfn>get matching cookies</dfn> given |cookie store| and |filter|:
+
+1. Let |cookies| be a new list.
+
+1. For each |stored cookie| in |cookie store|:
+
+  1. If [=match cookie=] with |stored cookie| and |filter| is true:
+
+    1. Append the result of [=serialize cookie=] given |stored cookie| to |cookies|.
+
+1. Return |cookies|.
+
+</div>
+
+### Commands ### {#module-storage-commands}
+
+#### The storage.getCookies Command ####  {#command-storage-getCookies}
+
+The <dfn export for=commands>storage.getCookies</dfn> command retrieves zero or more [=cookies=] which [=match cookie|match=] a set of provided parameters.
+
+<dl>
+   <dt>Command Type</dt>
+   <dd>
+    <pre class="cddl remote-cddl">
+        storage.GetCookies = {
+          method: "storage.getCookies",
+          params: storage.GetCookiesParameters
+        }
+
+        <!--
+        Modifications to this definition should be reflected in
+        `network.Cookie`, `network.SetCookieHeader`, and `storage.PartialCookie`.
+        -->
+        storage.CookieFilter = {
+          ? name: text,
+          ? value: network.BytesValue,
+          ? domain: text,
+          ? path: text,
+          ? size: js-uint,
+          ? httpOnly: bool,
+          ? secure: bool,
+          ? sameSite: network.SameSite,
+          ? expiry: js-uint,
+          Extensible,
+        }
+
+        storage.GetCookiesParameters = {
+          ? filter: storage.CookieFilter,
+          ? partition: storage.PartitionKey / browsingContext.BrowsingContext,
+        }
+    </pre>
+   </dd>
+   <dt>Result Type</dt>
+   <dd>
+    <pre class="cddl local-cddl">
+      storage.GetCookiesResult = {
+        cookies: [*network.Cookie],
+        partitionKey: storage.PartitionKey,
+      }
+    </pre>
+   </dd>
+</dl>
+
+<div algorithm="remote end steps for script.getCookies">
+The [=remote end steps=] with <var ignore>session</var> and |command parameters| are:
+
+1. Let |filter| be the value of the <code>filter</code> field of |command parameters|
+  if it is present or an empty [=/map=] if it isn't.
+
+1. Let |cookie domain| be the value of the <code>domain</code> field of |filter| if it is present or null if it isn't.
+
+1. Let |partition spec| be the value of the <code>partition</code> field of
+  |command parameters| if it is present or null if it isn't.
+
+1. Let |partition key| be the result of [=trying=] to [=expand a storage partition spec|expand=] |partition spec| using |cookie domain|.
+
+1. Let |store| be the result of [=trying=] to [=get the cookie store=] with
+  |partition key|.
+
+1. Let |cookies| be the result of [=get matching cookies=] with |store| and |filter|.
+
+1. Let |body| be a [=/map=] matching the <code>storage.GetCookiesResult</code> production,
+  with the <code>cookies</code> field set to |cookies| and the <code>partitionKey</code>
+  field set to |partition key|.
```

Comment:
The `partition key` field here means "this cookies were found for the given partition key". Let's say a browser supports partition keys `sourceOrigin` and `other_supported`, but not partition key `unsupported`. 
The user sends `storage.getCookies` with **partition**: `{sourceOrigin: 'foo', unsupported: 'baz'}`. The response would contain **partitionKey**: `{sourceOrigin: 'foo', other_supported: 'default_value', unsupported: 'baz'}`. 

I wonder if we should provide a way for users to understand that the `unsupported` partition key is not supported by the given browser.

Reply:
Basically, the question is if we want to provide the actual partition key for each returned cookie individually?

Reply:
@sadym-chromium I think I agree: it seems misleading for the user agent to reflect partition key properties that it does not recognize. This should be fairly straightforward to specify in the "expand a storage partition spec" algorithm (especially [if my latest change is appropriate](https://github.com/w3c/webdriver-bidi/pull/501#discussion_r1424814640)), though I could use some clarification regarding your follow-up question:

> Basically, the question is if we want to provide the actual partition key for each returned cookie individually?

As I understand it, the `storage.getCookies` command is only capable of returning cookies stored in one specific storage partition. From that perspective, it seems redundant for the remote end to return the partition key for each cookie individually.

Reply:
One per command rather than one per cookie seems good.